EKF/ekf_helper.cpp

```diff
@@ -824,27 +824,43 @@ void Ekf::get_ekf_ctrl_limits(float *vxy_max, float *vz_max, float *hagl_min, fl
 
 }
 
-bool Ekf::reset_imu_bias()
+void Ekf::resetIMUBias()
 {
-	if (_imu_sample_delayed.time_us - _last_imu_bias_cov_reset_us < (uint64_t)10e6) {
-		return false;
-
-	}
+	resetGyroBias();
+	resetAccelBias();
+}
 
+void Ekf::resetGyroBias()
+{
 	// Zero the delta angle and delta velocity bias states
 	_state.delta_ang_bias.zero();
-	_state.delta_vel_bias.zero();
 
 	// Zero the corresponding covariances and set
 	// variances to the values use for initial alignment
 	P.uncorrelateCovarianceSetVariance<3>(10, sq(_params.switch_on_gyro_bias * FILTER_UPDATE_PERIOD_S));
+}
+
+void Ekf::resetAccelBias()
+{
+	// Zero the delta angle and delta velocity bias states
+	_state.delta_vel_bias.zero();
+
+	// Zero the corresponding covariances and set
+	// variances to the values use for initial alignment
 	P.uncorrelateCovarianceSetVariance<3>(13, sq(_params.switch_on_accel_bias * FILTER_UPDATE_PERIOD_S));
-	_last_imu_bias_cov_reset_us = _imu_sample_delayed.time_us;
 
 	// Set previous frame values
-	_prev_dvel_bias_var = P.slice<3,3>(13,13).diag();
+	_prev_dvel_bias_var = P.slice<3,3>(13, 13).diag();
+}
 
-	return true;
+void Ekf::resetMagBias()
+{
+	// Zero the magnetometer bias states
+	_state.mag_B.zero();
+
+	// Zero the corresponding covariances and set
+	// variances to the values use for initial alignment
+	P.uncorrelateCovarianceSetVariance<3>(19, sq(_params.mag_noise));
```

Comment:
Would we need to reset `_saved_mag_bf_variance` as well?

Reply:
Sounds like it. 

FYI the use case is either magnetometer calibration updated or the device completely changed mis-use (failured).

Reply:
Why does `_saved_mag_bf_variance` include the D earth axis?

Reply:
https://github.com/PX4/PX4-ECL/blob/48a8992caf7a95e09a5e17235133894c072b96bd/EKF/ekf_helper.cpp#L1346-L1349

Reply:
Updated.

Reply:
I am honestly not sure, what is the right thing to do here. Should we reset it to 0, mag_noise or let it be. @bresch can you help us here?

Reply:
@bresch can you check this?